### PR TITLE
WW3 feature:  Langmuir turbulence parameterization

### LINKED ISSUE #2187: Add Langmuir parameterization to WW3
## Description
Add the Langmuir turbulence parameterization from ESCOMP which gives the option to compute Stokes drift diagnostics within WW3.

## Solution
A PR should be raised to bring this parameterization into the WW3 `dev/ufs-weather-model` branch.

## Alternatives
N/A.

## Related to
- needed by noaa-emc/ww3/pull/1034

### LINKED ISSUE #2198: rt.sh crashes with 'Permission denied' error while creating run directory on Gaea
## Description

When I run regression test on Gaea rt.sh crashes with 'Permission denied' error while creating run directory. Script tried to create directory under /gpfs/f5/epic/scratch, and I do not have permission to write to that directory.

### To Reproduce:

1. Run rt.sh on Gaea from a user account which is not member of the epic project.

## Output

```
+ PARTITION=c5
+ STMP=/gpfs/f5/epic/scratch
+ PTMP=/gpfs/f5/epic/scratch
+ SCHEDULER=slurm
+ mkdir -p /gpfs/f5/epic/scratch/Dusan.Jovic
mkdir: cannot create directory ‘/gpfs/f5/epic/scratch/Dusan.Jovic’: Permission denied
++ echo 'rt.sh error on line 897'
rt.sh error on line 897
++ cleanup

```

### PR DESCRIPTION
**Pull request by MatthewMasarik-NOAA**
## Commit Queue Requirements:
- [X] Fill out all sections of this template.
- [X] All sub component pull requests have been reviewed by their code managers.
- [X] Run the full Intel+GNU RT suite (compared to current baselines) on either Hera/Derecho/Hercules
- [X] Commit 'test_changes.list' from previous step
---
## Description:
Adds Langmuir turbulence parameterization developed by ESCOMP to allow computing Stokes drift diagnostics within the WW3 sub-component.

### Commit Message:
Add Langmuir turbulence parameterization for computing Stokes drift diagnostics.  Authors: @qingli411, @alperaltuntas
```
  * WW3 - Langmuir turbulence parameterization
```

### Priority:
* Normal.

### UFSWM:
* Closes #2187

### Sub component Pull Requests:
* WW3:  needed by noaa-emc/ww3/pull/1034

### UFSWM Blocking Dependencies:
* None

---
### Regression Test Changes (Please commit test_changes.list):
* No Baseline Changes.

[2024-03-13.RegressionTests_hera.log.txt](https://github.com/ufs-community/ufs-weather-model/files/14619315/2024-03-13.RegressionTests_hera.log.txt), test_changes.list commited:  517e05d

### Input data Changes:
* Updated input data.  

New `mod_def`'s for all RTs are the updates to the input data.  Directory location containing new `mod_def`s:  `/scratch1/NCEPDEV/climate/Matthew.Masarik/waves/share/ufs/moddefs/WW3_input_data_20240214`.

### Library Changes/Upgrades:
* No Updates
  
---

## Testing Log:
- RDHPCS
  - [x] Hera
  - [x] Orion
  - [x] Hercules
  - [x] Jet
  - [x] Gaea
  - [x] Derecho
- WCOSS2
  - [x] Dogwood/Cactus
  - [ ] Acorn
- [x] CI
- [ ] opnReqTest (complete task if unnecessary)

### CODE REVIEW

**Comment by jkbk2004** [comment]
@MatthewMasarik-NOAA can you sync up branch?

**Comment by MatthewMasarik-NOAA** [comment]
Hi @jkbk2004, looks like theres a merge conflict.  Let me look into it

**Comment by MatthewMasarik-NOAA** [comment]
Hi @jkbk2004, the conflict is just in rt.sh. The difference is INPUTDATA_ROOT_WW3 which I set to the directory with the updated mod_defs.  @JessicaMeixner-NOAA had advised this.  Should I resolve this by committing my version of rt.sh, or revert it to the normal state?

**Comment by jkbk2004** [comment]
@MatthewMasarik-NOAA I rsynced up to WW3_input_data_20220624 on hera. Do you like to have WW3_input_data_20240214 in baseline input directory? I backed up WW3_input_data_20220624. So no problem to go with WW3_input_data_20240214.

**Comment by MatthewMasarik-NOAA** [comment]
@jkbk2004, yes that sounds great.  please add WW3_input_data_20240214 to the baseline input directory.

So for my side, I should set `INPUTDATA_ROOT_WW3=${INPUTDATA_ROOT}/WW3_input_data_20240214` in `rt.sh` to resolve the conflict, is that correct?

**Comment by DeniseWorthen** [comment]
@jkbk2004 Do not over-write the existing  WW3_input_data_20220624. You need to create a new dated WW3_input_data inside of input-data for the new mod_def files.

**Comment by jkbk2004** [comment]
> @jkbk2004, yes that sounds great. please add WW3_input_data_20240214 to the baseline input directory.
> 
> So for my side, I should set `INPUTDATA_ROOT_WW3=${INPUTDATA_ROOT}/WW3_input_data_20240214` in `rt.sh` to resolve the conflict, is that correct?

@MatthewMasarik-NOAA /scratch2/NAGAPE/epic/UFS-WM_RT/NEMSfv3gfs/input-data-20221101/WW3_input_data_20240214 is there now. Go ahead to set as INPUTDATA_ROOT_WW3=${INPUTDATA_ROOT}/WW3_input_data_20240214. @BrianCurtis-NOAA @zach1221 @FernandoAndrade-NOAA FYI

**Comment by jkbk2004** [comment]
> WW3_input_data_20220624

WW3_input_data_20220624 is intact on hera.

**Comment by DeniseWorthen** [comment]
Please make sure that the original WW3_input_data_20220624/ is in place. If I look on Hera, those files have the current day's timestamp.

**Comment by MatthewMasarik-NOAA** [comment]
> @MatthewMasarik-NOAA /scratch2/NAGAPE/epic/UFS-WM_RT/NEMSfv3gfs/input-data-20221101/WW3_input_data_20240214 is there now. Go ahead to set as INPUTDATA_ROOT_WW3=${INPUTDATA_ROOT}/WW3_input_data_20240214.

@jkbk2004, I'll go ahead with that

**Comment by MatthewMasarik-NOAA** [comment]
@jkbk2004, I just resolved the conflict and updated my branch.  I'm not completely sure if the FV3,MOM6 updates come in OK.  please let me know if not

**Comment by jkbk2004** [comment]
> @jkbk2004, I just resolved the conflict and updated my branch. I'm not completely sure if the FV3,MOM6 updates come in OK. please let me know if not

looks good now.

**Comment by jkbk2004** [comment]
hera is slow today. But I submitted jobs. @zach1221 @FernandoAndrade-NOAA please, pick up other machines. @BrianCurtis-NOAA FYI: this pr is ready.

**Comment by jkbk2004** [comment]
@FernandoAndrade-NOAA can you reflect to this pr for the STMP/PTMP gaea line update needed to cover https://github.com/ufs-community/ufs-weather-model/issues/2198 ? We need to update the ecflow version number for hera as well.

**Comment by jkbk2004** [comment]
looks like we need to resolve conflicts on rt.sh in this pr. @MatthewMasarik-NOAA we don't need to update rt.sh except https://github.com/MatthewMasarik-NOAA/ufs-weather-model/blob/feature/langmuir-parameterization/tests/rt.sh#L938, right?

**Comment by FernandoAndrade-NOAA** [comment]
> looks like we need to resolve conflicts on rt.sh in this pr. @MatthewMasarik-NOAA we don't need to update rt.sh except https://github.com/MatthewMasarik-NOAA/ufs-weather-model/blob/feature/langmuir-parameterization/tests/rt.sh#L938, right?

the conditionals and load for rocoto and ecflow were removed with this PRs rt.sh changes as well

File: tests/2024-03-13.test_changes.list.txt
Comment (BrianCurtis-NOAA) [inline tests/2024-03-13.test_changes.list.txt:1]: This file should be renamed "test_changes.list"

**Review by BrianCurtis-NOAA — CHANGES_REQUESTED** [review]

File: tests/2024-03-13.test_changes.list.txt
Comment (DeniseWorthen) [reply tests/2024-03-13.test_changes.list.txt:1]: And can someone do white-space clean up on rt.sh?

**Comment by FernandoAndrade-NOAA** [comment]
> looks like we need to resolve conflicts on rt.sh in this pr. @MatthewMasarik-NOAA we don't need to update rt.sh except https://github.com/MatthewMasarik-NOAA/ufs-weather-model/blob/feature/langmuir-parameterization/tests/rt.sh#L938, right?

I restored rt.sh, updated the WW3 input date and added the Gaea paths update for stmp and ptmp for users outside of the epic group.

**Comment by MatthewMasarik-NOAA** [comment]
> looks like we need to resolve conflicts on rt.sh in this pr. @MatthewMasarik-NOAA we don't need to update rt.sh except https://github.com/MatthewMasarik-NOAA/ufs-weather-model/blob/feature/langmuir-parameterization/tests/rt.sh#L938, right?

@jkbk2004 
this was the only line I thought needed to be updated: https://github.com/MatthewMasarik-NOAA/ufs-weather-model/blob/055b13ffbe0e49abbcf8d4e503ee617a8e78e5be/tests/rt.sh#L990

**Comment by MatthewMasarik-NOAA** [comment]
Regarding rt.sh, test_changes.list, and whitespace -- 
please let me know if I should make these updates to the branch, or if these comments are for CMs

**Comment by FernandoAndrade-NOAA** [comment]
> > looks like we need to resolve conflicts on rt.sh in this pr. @MatthewMasarik-NOAA we don't need to update rt.sh except https://github.com/MatthewMasarik-NOAA/ufs-weather-model/blob/feature/langmuir-parameterization/tests/rt.sh#L938, right?
> 
> @jkbk2004 this was the only line I thought needed to be updated: https://github.com/MatthewMasarik-NOAA/ufs-weather-model/blob/055b13ffbe0e49abbcf8d4e503ee617a8e78e5be/tests/rt.sh#L990

That's the line he was pointing to before I restored and resynced rt.sh, sorry about that confusion.

File: tests/2024-03-13.test_changes.list.txt
Comment (DeniseWorthen) [reply tests/2024-03-13.test_changes.list.txt:1]: Whitespace looks OK now. I'm not sure what had happened to it.

**Comment by zach1221** [comment]
Ok let's start the merge process.

**Comment by JessicaMeixner-NOAA** [comment]
WW3 is merged: https://github.com/NOAA-EMC/WW3/commit/d9b3172f4197c65d471662c6952a668152d71230 
@MatthewMasarik-NOAA

**Review by jkbk2004 — APPROVED** [review]

**Review by BrianCurtis-NOAA — APPROVED** [review]

**Pull request merged**